src/Model/APContact.php

```diff
@@ -539,4 +538,21 @@ private static function unarchiveInbox(string $url, bool $shared)
 
 		HTTPSignature::setInboxStatus($url, true, $shared);
 	}
+
+	public static function isRelay(array $apcontact): bool
+	{
+		if ($apcontact['nick'] != 'relay') {
+			return false;
+		}
+
+		if ($apcontact['type'] == 'Application') {
+			return true;
+		}
+
+		if (in_array($apcontact['type'], ['Group', 'Service']) && ($apcontact['nick'] == 'relay') && is_null($apcontact['outbox'])) {
```

Comment:
`$apcontact['nick'] == 'relay'` is always `true` at this point thanks to the condition on line 544, which means it can be removed from this condition.
```suggestion
		if (in_array($apcontact['type'], ['Group', 'Service']) && is_null($apcontact['outbox'])) {
```

Reply:
Yeah. In fact this is part of some other unfinished work to support more relay types.